src/types/typeobject.rs

```diff
@@ -11,38 +13,141 @@ pub struct PyType(PyAny);
 pyobject_native_type_core!(PyType, pyobject_native_static_type_object!(ffi::PyType_Type), #checkfunction=ffi::PyType_Check);
 
 impl PyType {
-    /// Creates a new type object.
+    /// Deprecated form of [`PyType::new_bound`].
     #[inline]
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`new` will be replaced by `new_bound` in a future PyO3 version"
```

Comment:
For the other types we have included the type name in the note, but I wouldn't consider that super important.

Reply:
Ah good catch, I'll fixup, this is because this code is now so old it misses out a lot of our current conventions 👍